### ISSUE TITLE
Python: Bug: extension_data doesn't seem to propagate in chat completions API

### ISSUE THREAD

**Issue by barucoh** [microsoft/semantic-kernel#9456]
**Describe the bug**
I want to use the `parallel_tool_calls=False` flag to only call a single function at a time.
Trying to use it with `extension_data` doesn't seem to work.
I put a breakpoint in the `openai._base_client.py : AsyncAPIClient._request` function to see what is exactly being sent to the API and the flag is not there.

**To Reproduce**

```
        settings: OpenAIChatPromptExecutionSettings = (
            self.kernel.get_prompt_execution_settings_from_service_id(
                service_id="service_name"
            )
        )
        settings.extension_data = {"parallel_tool_calls": False}

        arguments = KernelArguments(settings=settings)

        ...

        chat_function = self.kernel.add_function(
            prompt="{{$chat_history}}",
            plugin_name="ChatBot",
            function_name="Chat",
        )
        result = await self.kernel.invoke(chat_function, arguments=arguments)
```

**Platform**
 - OS: Windows
 - IDE: VS Code
 - Language: Python
 - Source: semantic-kernel ==1.1.2

**Comment by evchaki** [comment]
@barucoh thanks for bringing this up. We will take a look and see what we can do to support this. @moonbox3

**Comment by moonbox3** [comment]
Hi @barucoh, thanks for reporting this. The `extension_data` dict looks a bit misleading -- it's supposed to be a way to propagate kwargs into specific prompt execution settings where the keys/attributes already exist. We understand you'd like a way to handle the `parallel_tools_calls` attribute. We've added this in via the following PR (which will be released today): #9479
